integration-libs/opf/checkout/components/opf-checkout-payment-wrapper/opf-checkout-payment-wrapper.service.spec.ts

```diff
@@ -1 +1,294 @@
+        },
+      } as RouterState)
+    );
+
+    TestBed.configureTestingModule({
+      providers: [
+        OpfCheckoutPaymentWrapperService,
+        { provide: OpfCheckoutFacade, useValue: opfCheckoutFacadeMock },
+        { provide: OpfOtpFacade, useValue: opfOtpFacadeMock },
+        {
+          provide: OpfResourceLoaderService,
+          useValue: opfResourceLoaderServiceMock,
+        },
+        { provide: UserIdService, useValue: userIdServiceMock },
+        { provide: ActiveCartService, useValue: activeCartServiceMock },
+        { provide: RoutingService, useValue: routingServiceMock },
+        { provide: GlobalMessageService, useValue: globalMessageServiceMock },
+        { provide: OpfOrderFacade, useValue: opfOrderFacadeMock },
+        { provide: OpfService, useValue: opfServiceMock },
+      ],
+    });
+
+    service = TestBed.inject(OpfCheckoutPaymentWrapperService);
+  });
+
+  it('should retrieve renderPaymentMethodEvent$', (done) => {
+    const mockRenderPaymentMethodEvent = { isLoading: false, isError: false };
+    service['renderPaymentMethodEvent$'].next(mockRenderPaymentMethodEvent);
+
+    service.getRenderPaymentMethodEvent().subscribe((event) => {
+      expect(event).toEqual(mockRenderPaymentMethodEvent);
+      done();
+    });
+  });
+
+  it('should initiate payment successfully and render payment gateway', (done) => {
+    const mockPaymentOptionId = 123;
+    const mockOtpKey = 'otpKey';
+    const mockUserId = 'userId';
+    const mockCartId = 'cartId';
+    const mockPaymentSessionData: PaymentSessionData = {
+      dynamicScript: {
+        html: '<html></html>',
+        jsUrls: [
+          { url: 'script.js', type: PaymentDynamicScriptResourceType.SCRIPT },
+        ],
+        cssUrls: [
+          { url: 'styles.css', type: PaymentDynamicScriptResourceType.STYLES },
+        ],
+      },
+    };
+
+    opfCheckoutFacadeMock.initiatePayment.and.returnValue(
+      of(mockPaymentSessionData)
+    );
+    opfOtpFacadeMock.generateOtpKey.and.returnValue(of({ value: mockOtpKey }));
+    userIdServiceMock.getUserId.and.returnValue(of(mockUserId));
+    activeCartServiceMock.getActiveCartId.and.returnValue(of(mockCartId));
+    routingServiceMock.getRouterState.and.returnValue(
+      of({ state: { semanticRoute: 'checkoutReviewOrder' } } as RouterState)
+    );
+    routingServiceMock.getFullUrl.and.returnValue('http://sap.com');
+    opfServiceMock.updateOpfMetadataState.and.stub();
+    opfResourceLoaderServiceMock.loadProviderResources.and.returnValue(
+      Promise.resolve()
+    );
+    spyOn(service, 'renderPaymentGateway').and.callThrough();
+
+    service.initiatePayment(mockPaymentOptionId).subscribe(() => {
+      expect(opfCheckoutFacadeMock.initiatePayment).toHaveBeenCalledWith({
+        otpKey: mockOtpKey,
+        config: {
+          configurationId: mockPaymentOptionId.toString(),
+          cartId: mockCartId,
+          resultURL: 'http://sap.com',
```

Comment:
Can we create a constant at the top of the file to store a mock URL, allowing us to reuse this variable throughout the code?

Reply:
Changed